x/datapool/keeper/pool_distribute.go

```diff
@@ -32,6 +32,11 @@ func (k Keeper) SetSalesHistory(ctx sdk.Context, salesHistory *types.SalesHistor
 		salesHistory.Round,
 		salesHistory.SellerAddress,
 	)
+	k.SetSalesHistoryByKey(ctx, key, salesHistory)
+}
+
+// SetSalesHistoryByKey stores sales history.
+func (k Keeper) SetSalesHistoryByKey(ctx sdk.Context, key []byte, salesHistory *types.SalesHistory) {
```

Comment:
I just wonder why you made this function splitted which stores history by key. Is there any specific reason?!

Reply:
It is a residue created while editing the code.
I removed this function. :)